ext/json/json.c

```diff
@@ -222,19 +232,35 @@ PHP_FUNCTION(json_encode)
 	zval *parameter;
 	php_json_encoder encoder;
 	smart_str buf = {0};
+	smart_str indent_str = {0};
+	zend_string *input_indent_str = 0;
 	zend_long options = 0;
 	zend_long depth = PHP_JSON_PARSER_DEFAULT_DEPTH;
+	zend_long indent = PHP_JSON_ENCODER_DEFAULT_INDENT;
 
-	ZEND_PARSE_PARAMETERS_START(1, 3)
+	ZEND_PARSE_PARAMETERS_START(1, 4)
 		Z_PARAM_ZVAL(parameter)
 		Z_PARAM_OPTIONAL
 		Z_PARAM_LONG(options)
 		Z_PARAM_LONG(depth)
+		Z_PARAM_STR_OR_LONG(input_indent_str, indent)
 	ZEND_PARSE_PARAMETERS_END();
 
 	php_json_encode_init(&encoder);
 	encoder.max_depth = (int)depth;
+
+	if (input_indent_str) {
+		indent_str.s = input_indent_str;
+	} else {
```

Comment:
This should check for `indent < 0` and throw `zend_argument_value_error` in that case.

Reply:
Added the validation and accompanying test!